### DISCUSSION TITLE
How to help with notorious PR review problem

### DISCUSSION THREAD

**Discussion by Brackets-Coder** [TurboWarp/extensions#1721] [Q&A] [answered]
Hello everyone, 
I'm mildly concerned about this repository's notorious issue with impatience on the part of some contributors (including myself, I admit) and tardiness on the part of some reviewers. As of the time of writing, there are 110 open pull requests, the oldest dating back to June 1 of 2023 - 16 months ago. 

I don't know if some of these extensions were never meant to be reviewed because they don't follow the requirements or something like that, but I'm wondering if some of these older extensions may become slightly outdated (although not obsolete) before they can be approved.

I guess there's two points I'm trying to ask here:

1. Can someone explain to me what I'm not understanding about this problem? I'm aware that most of the contributors and maintainers have lives outside of Turbowarp (mostly full-time jobs or academic studies), and I fully understand that perspective, but please tell me if there's any other details that I'm missing or not understanding.

2. Is there anything I can do to help speed up the process? What's required to review and approve extensions, etc? I'm entirely willing to help tackle this issue, but I don't exactly know what I can do to help.

Thank you for understanding and helping.

**Comment by CST1229** [comment]
> 1. Can someone explain to me what I'm not understanding about this problem? I'm aware that most of the contributors and maintainers have lives outside of Turbowarp (mostly full-time jobs or academic studies), and I fully understand that perspective, but please tell me if there's any other details that I'm missing or not understanding.

alongside the maintainers having lives, there is also exactly one person who can merge pull requests: garbomuffin (and maybe cubester but i don't know if he's willing to merge prs yet)

**Comment by GarboMuffin** [comment, marked as answer]
I think it's clear at this point that we need to move to something more like Scratch Addons' two-approvals-then-merge system since having me review things just doesn't work

@DNin01 @CST1229 @Xeltalliv you three have sent consistently high quality pull requests in the past. if you'd like I can invite you each with write access here

**Reply by GarboMuffin** [reply]
same offer for @David-Orangemoon @LilyMakesThings and maybe any other prior contributors that ask nicely

**Reply by DNin01** [reply]
I'm not the most experienced, but I can help.

**Reply by David-Orangemoon** [reply]
Ive moved away from tw and scratch extensions for a bit srry.

**Reply by CST1229** [reply]
> I'm not the most experienced, but I can help.

Same.

**Reply by GarboMuffin** [reply]
I will update the contributing document with new guidelines sometime soon.

For now. If there's some PR that's interesting and you like it and it works and there's no obvious security problems, leave a review on it. Once it gets 2 maintainer approvals the merge button should be available. You can press it if you want to.

If this works well it would be nice to adopt a similar model for the other repositories

**Reply by GarboMuffin** [reply]
I did label all open PRs. Would like to make some templates

**Reply by hammouda101010** [reply]
there should be more reviewers in this repo. a new extension each month?! that's insanly slow for so many new extensions.  ☹️

**Reply by DNin01** [reply]
> there should be more reviewers in this repo. a new extension each month?! that's insanly slow for so many new extensions.  ☹️ 

There _are_ only so many kinds of extensions we can make, though. But I think we could use some help with triaging and reviews on the issues and PRs here.

**Comment by CubesterYT** [comment]
Only Garbo can merge PRs right now, and I only have the power to just manage, review, etc. But I fully support Garbo's thoughts on the Scratch Addons system. So that more extensions are reviewed and merged.

**Reply by Brackets-Coder** [reply]
I agree as well, but I'd like to be able to help keep traffic flowing more smoothly. Is there any specific process on reviewing? Is there anything I can do to help the situation? Must I be assigned specific access to proceed further? These are all questions I have. Thanks for the help, I've learned a lot already.

**Comment by Brackets-Coder** [comment]
I think it might also be helpful if we use PR labels more so you can see what category a PR is at a glance, helping merge more critical things faster with more priority (e.g., fix extension bug vs. new extension vs. dependency update)

**Comment by Brackets-Coder** [comment]
Another thing we should consider is a client-side search, sort, and filter system for the extensions homepage file, especially when with these changes there will likely be a significant increase in extension releases.

**Comment by Brackets-Coder** [comment]
it looks like this problem is getting better. I guess I can close this now?
